### ISSUE TITLE
Reordering ACL rules inside the ACL list - does not apply the rule order inside the VR

### ISSUE THREAD

**Issue by s-novakovic** [apache/cloudstack#9398]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI (maybe also API)
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1.0-SNAPSHOT
~~~

##### SUMMARY

Using DRAG&DROP to arrange ACL rules order inside ACL list, doesn't apply rules inside the VR (the ACL list is already attached to a VPC network) The workaround is to change ACL list for this network to another ACL list and then go back and reapply previously used  ACL list. 

##### STEPS TO REPRODUCE

Before changing  the order of rules outgoing traffic to port 80 is allowed. 

![image](https://github.com/user-attachments/assets/4de80622-7899-4005-a316-dad8704968ff)

After putting the DENY rule above the port 80 rule, the outgoing traffic to port 80  is still allowed - see the screenshots bellow 

 
![image](https://github.com/user-attachments/assets/3b81de55-1275-4966-bdf9-94124f0f2d1b)

![image](https://github.com/user-attachments/assets/1ca2c4b9-c7a9-4884-a646-9dcc9274f4e6)

##### EXPECTED RESULTS

~~~
At least - show the message in the UI (API output) that the ACL list needs to be reapplied.
~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by DaanHoogland** [comment]
@s-novakovic did you try restarting the network with cleanup?

**Comment by btzq** [comment]
@DaanHoogland i think this issue is very similar to the one we discussed in https://github.com/apache/cloudstack/issues/8865

**Comment by DaanHoogland** [comment]
ok, in that case can you check with the new RC for 4.19.1? #9119 could have solved it.

**Comment by s-novakovic** [comment]
> @s-novakovic did you try restarting the network with cleanup?

Yes @DaanHoogland after restarting the network the rules get applied and it works as expected. Maybe we should add a message in the UI/API output, something like "you might need to restart the network for the rules to get applied" or something similar. I we'll check to behavior in the new RC and we'll let you know. Thank you

**Comment by DaanHoogland** [comment]
Well @s-novakovic , we could add a message, but I think your initial expectations are not wrong. I just saw you had used a snapshot 4.19.1 version. Can you check if that included #9119 ?

**Comment by s-novakovic** [comment]
@DaanHoogland, I have checked the #9119 and it only modifies 2 UI-based files (.js and .vue), so it can't be fixing it...?
However, I have managed to somehow upgrade from 4.19.1.0-snapshot to the 4.19.1.0-RC2 and it still doesn't work.

Will try a clean RC2 install and let you know if there are any changes.

**Comment by s-novakovic** [comment]
I've managed to rest this on the RC2/final 4.19.1.0 version - and the issue is still there. It seems that only the DB records get updated (the order of rules for the ACL list) - but we never check if the ACL list is already attached to a network - and thus we never update the rules inside the VR.

It would be good if we do a check "if ACL is used/attached to a network --> do some nice message that network needs to be restarted (there is no cleanup option for the VPC network, so it's not disruptive and it's a quick action).

**Comment by Curverneur** [comment]
Hi, we noticed this behaviour as well and did some tests. Version: 4.22.0

**Example:**
<img width="1617" height="882" alt="Image" src="https://github.com/user-attachments/assets/2e339627-f997-48bb-9d13-5164599f03b8" />

```bash
root@r-1948-VM:~# iptables -t mangle -L -n -v --line-numbers
# [...]
Chain ACL_OUTBOUND_eth2 (1 references)
num   pkts bytes target     prot opt in     out     source               destination
1        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            224.0.0.18
2        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            10.0.0.0/8
3        0     0 ACCEPT     6    --  *      *       0.0.0.0/0            192.168.1.0/24       tcp dpt:443
4       11   836 DROP       0    --  *      *       0.0.0.0/0            0.0.0.0/0
5        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            225.0.0.50
# [...]
```
  
  
**How it doesn't work:**
If I try to swap the rule `10.0.0.0/8` with `192.168.1.0/24` by using the "move" icon (drag&drop), the page refreshes asap and the ui shows changed rule numbers:
<img width="58" height="54" alt="Image" src="https://github.com/user-attachments/assets/3dab9f34-eb26-40ca-bbdf-086c973b3eea" />

<img width="1605" height="875" alt="Image" src="https://github.com/user-attachments/assets/2bc44808-7e08-455f-b250-5e7c3dbaed4a" />

Checking the VirtualRouter indicates no change:
````bash
root@r-1948-VM:~# iptables -t mangle -L -n -v --line-numbers
# [...]
Chain ACL_OUTBOUND_eth2 (1 references)
num   pkts bytes target     prot opt in     out     source               destination
1        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            224.0.0.18
2        4   264 ACCEPT     0    --  *      *       0.0.0.0/0            10.0.0.0/8
3        0     0 ACCEPT     6    --  *      *       0.0.0.0/0            192.168.1.0/24       tcp dpt:443
4       98  7448 DROP       0    --  *      *       0.0.0.0/0            0.0.0.0/0
5        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            225.0.0.50
# [...]
````

Reverting the change in the ui to get back to the starting position:
<img width="1617" height="884" alt="Image" src="https://github.com/user-attachments/assets/771d0059-6736-412f-ad95-e390817afbad" />
  
  
**How it works:**
Changing the rule number of `192.168.1.0/24` from 4 to 2 to move it before`10.0.0.0/8` using the edit button:
<img width="47" height="39" alt="Image" src="https://github.com/user-attachments/assets/7a2921d1-3e94-4dfb-b3f4-16f51293f1c9" />

<img width="518" height="472" alt="Image" src="https://github.com/user-attachments/assets/d7aca931-1899-41c8-b968-0ed6ae97a42c" />

--> you notice, that the page takes longer to save
<img width="1604" height="869" alt="Image" src="https://github.com/user-attachments/assets/4d23dc2e-130c-4993-bda4-8cb13052ae8b" />
--> the rule numbers changed in the VirtualRouter:
```bash
root@r-1948-VM:~# iptables -t mangle -L -n -v --line-numbers
# [...]
Chain ACL_OUTBOUND_eth2 (1 references)
num   pkts bytes target     prot opt in     out     source               destination
1        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            224.0.0.18
2        0     0 ACCEPT     6    --  *      *       0.0.0.0/0            192.168.1.0/24       tcp dpt:443
3        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            10.0.0.0/8
4       21  1596 DROP       0    --  *      *       0.0.0.0/0            0.0.0.0/0
5        0     0 ACCEPT     0    --  *      *       0.0.0.0/0            225.0.0.50
# [...]
```

**Summary:**
Changing the rule number of an ACL Rule (live) only works by using the edit button. Changes by Terraform (if rule_number is set) are applied to the VirtualRouter as expected.

**Comment by shwstppr** [comment]
Fixed by #12525
